### ISSUE TITLE
Decompiler: NullCoalescingAssignmentPass false-raises manual property/indexer null assignments to ??=

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1746]
## Defect

`NullCoalescingAssignmentPass` raises a manual property/indexer null-test assignment into `??=` even when the IL lacks the compiler spill shape that proves source `??=`. The result is `Full` decompiled C# that recompiles to a different opcode stream.

This is a false-positive raise, not a readability preference: compile-back fidelity reports opcode diffs for the manual forms.

## Repro specimen

```csharp
using System.Collections.Generic;

public class NullAssignSpecimens
{
    public string? Prop { get; set; }

    public string ManualPropNullAssign(string fallback)
    {
        if (Prop is null)
            Prop = fallback;
        return Prop;
    }

    public string ManualIndexerNullAssign(Dictionary<int, string> map, int key, string fallback)
    {
        if (map[key] is null)
            map[key] = fallback;
        return map[key];
    }

    public string IndexerNullAssign(Dictionary<int, string> map, int key, string fallback)
    {
        map[key] ??= fallback;
        return map[key];
    }
}
```

Build it as a Release class library (net11.0 here, but the shape is the important part), then run:

```bash
dotnet build src/dotnet-inspect -c Release

dotnet artifacts/bin/dotnet-inspect/release/dotnet-inspect.dll \
  member NullAssignSpecimens ManualPropNullAssign \
  --library /tmp/dvOps/bin/Release/net11.0/dvOps.dll \
  -S "Decompiled Source" --bare

dotnet artifacts/bin/dotnet-inspect/release/dotnet-inspect.dll \
  member NullAssignSpecimens ManualIndexerNullAssign \
  --library /tmp/dvOps/bin/Release/net11.0/dvOps.dll \
  -S "Decompiled Source" --bare

dotnet run --project tools/DecompilerHarness -c Release -- \
  /tmp/dvOps/bin/Release/net11.0/dvOps.dll \
  --fidelity-check --max-examples 20
```

## Actual output

Both manual methods decompile as `??=`:

```csharp
public string? ManualPropNullAssign(string? fallback)
{
    Prop ??= fallback;
    return Prop;
}

public string ManualIndexerNullAssign(System.Collections.Generic.Dictionary<int, string> map, int key, string fallback)
{
    map[key] ??= fallback;
    return map[key];
}
```

Harness evidence:

```text
COMPILE-BACK over 45 rendered methods (40 Full)

  exact opcode match : 34 (75.56%)
  opcode diff (Full) : 2 — recompiled to a different stream (the docket)
  context-build fail : 0 — could not emit the type skeleton
  recompile fail     : 4 — skeleton + body did not compile
  recompile-fail by code:
    CS1061: 4

Opcode-diff examples (Full):
  NullAssignSpecimens::ManualPropNullAssign
    orig : ldarg call brtrue ldarg ldarg call ldarg call ret
    recmp: ldarg call brtrue ldarg ldarg dup stloc call ldarg call ret
  NullAssignSpecimens::ManualIndexerNullAssign
    orig : ldarg ldarg callvirt brtrue ldarg ldarg ldarg callvirt ldarg ldarg callvirt ret
    recmp: ldarg stloc ldarg stloc ldloc ldloc callvirt brtrue ldloc ldloc ldarg dup stloc callvirt ldarg ldarg callvirt ret
```

The actual `map[key] ??= fallback` source has the spill shape:

```text
IL_0000: ldarg.1
IL_0001: stloc.0
IL_0002: ldarg.2
IL_0003: stloc.1
IL_0004: ldloc.0
IL_0005: ldloc.1
IL_0006: callvirt     System.Collections.Generic.Dictionary<int, string>::get_Item(int)
IL_000B: brtrue.s     IL_0017
IL_000D: ldloc.0
IL_000E: ldloc.1
IL_000F: ldarg.3
IL_0010: dup
IL_0011: stloc.2
IL_0012: callvirt     System.Collections.Generic.Dictionary<int, string>::set_Item(int, string)
```

The manual indexer form has no receiver/index/value spills:

```text
IL_0000: ldarg.1
IL_0001: ldarg.2
IL_0002: callvirt     System.Collections.Generic.Dictionary<int, string>::get_Item(int)
IL_0007: brtrue.s     IL_0011
IL_0009: ldarg.1
IL_000A: ldarg.2
IL_000B: ldarg.3
IL_000C: callvirt     System.Collections.Generic.Dictionary<int, string>::set_Item(int, string)
```

`--dump 'NullAssignSpecimens::ManualPropNullAssign' --diff` shows the bad raise occurs at `null-coalescing-assignment` after `structuring` turns the diamond into an `IfStatement`.

## Expected

The property/indexer `??=` raise should decline the direct setter form unless the compiler-specific spills prove source `??=`. The manual forms should stay as an `if` (or degrade honestly if the pass cannot distinguish a shape), while true property/indexer `??=` keeps raising.

**Comment by richlander** [comment]
Burndown runner claim: I am claiming this candidate as a runner-sized semantic-wrong-`Full` decompiler row. I have not started code changes; pausing here for maintainer instructions before implementation.

**Comment by richlander** [comment]
Runner update: #1746 is now in review via #1846. The PR fixes the instance property/indexer manual-null-assignment false raise, includes compiled-fixture negatives, and documents the static-property sibling follow-up as #1845.

**Comment by richlander** [comment]
Curator update: tracked as Done in semantic-wrong-`Full` burndown #1859 via merged #1846.
